src/borg/locking.py

```diff
@@ -158,6 +161,41 @@ def is_locked(self):
     def by_me(self):
         return os.path.exists(self.unique_name)
 
+    def kill_dead_lock(self):
+        for name in os.listdir(self.path):
+
+            try:
+                host_pid, thread = name.rsplit('-', 1)
+                host, pid = host_pid.rsplit('.', 1)
+            except ValueError:
+                # Malformed lock name? Or just soem new format we don't understand?
+                # It's safer to just exit
+                return False
+
+            if host != _hostname:
+                return False
+
+            try:
+                # This may not work in Windows.
+                os.kill(int(pid), 0)
+                return False
+            except OSError:
+                pass
```

Comment:
Well, remember there was a discussion on the itrc from two people already that wanted to run their serve process under different gids because some of them were local and some were remote.
There was some trickery with umask and sgid. This is the case that would get caught here - we'll not be able to kill a more privileged process potentially, but we will be able to remove the lock file I imagine, due to umask.